### ISSUE TITLE
Post Status is not updating in Gutenberg Editor

### ISSUE THREAD

**Issue by mavinothkumar** [WordPress/gutenberg#34958]
### Description

Hi,

In Gutenberg Editor, the post status (original_post_status) is not updating as expected. The hidden input **original_post_status** value is not properly update based on the post status update action. Whenever, we publish the post first time and with out refreshing the post the original_post_status value is not updating.

### Step-by-step reproduction instructions

1. Go to Posts | Add New Post.
2. Enter content and block.
3. Publish the post.

### Screenshots, screen recording, code snippet

**On adding the new Post and before it get published.**
1. Status: auto-draft (expected result: auto-draft)

![before-post-publish](https://user-images.githubusercontent.com/1659203/133992020-90b0e527-7f37-4066-bd0c-9474dbc1fd5a.png)

**After publishing the post**
1. Status: auto-draft (expected result: publish)
![after-post-published](https://user-images.githubusercontent.com/1659203/133992029-48b1df43-57d4-437c-bb72-7fbd443c1e92.png)

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by skorasaurus** [comment]
Hi, 

I'm able to reproduce this in Gutenberg 11.6 but I'm not quite sure what original_post_value 's use without spending more time on the issue. 

For context, here it is mentioned in source https://github.com/WordPress/wordpress-develop/blame/077504e755993f4385e583fb71f387e8829c1ecc/src/wp-admin/includes/post.php#L2423 and https://github.com/WordPress/wordpress-develop/blob/c00ba50d598abb1cd8aaba057da9555a452ffcee/src/wp-admin/edit-form-advanced.php#L474

**Comment by Thelmachido** [comment]
I am not able to reproduce this, on a test site: 

1. I created a post, and checked the source code before publishing. The `original_post_status` was `auto_draft `as shown in the screenshot below. 

<img width="1183" alt="Untitled 2" src="https://user-images.githubusercontent.com/41231764/186841584-8f7525e5-8027-4e3d-8055-155d95e36960.png">


3. After publishing the post I checked the source code without refreshing the page and the original_post_status was `publish `as shown below. 
<img width="976" alt="Untitled 3" src="https://user-images.githubusercontent.com/41231764/186841642-8879a5ed-ad25-4d29-b6ab-67cacb01fe9e.png">

WordPress 6.0.1 

Gutenberg 13.9.0 

Theme TT2

Closing this issue
